doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+     `Cumulative`      :attr:`universes(cumulative)`
+     `NonCumulative`   :attr:`universes(noncumulative)`
+     `Private`         :attr:`private(matching)`
+     `Program`         :attr:`program`
+     ================  ================================
+
+     Attributes appear in the HTML documentation in blue or gray boxes
+     after the label "Attribute".  In the pdf, they appear after the
+     boldface label "Attribute:".  Attributes are listed in the
+     :ref:`attribute_index`.
+
+     .. warn:: Unsupported attribute
+
+        This warning is fatal by default.  It is caused by using a
+        command with some attribute it does not understand.
+
+.. _flags-options-tables:
+
+Flags, Options and Tables
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Besides the use of :term:`attributes <attribute>` to control the
+behavior of a single command, it is possible to control the behavior
+of all the following commands using flags, options and tables:
+
+* A :gdef:`flag` has a boolean value, such as :flag:`Asymmetric Patterns`.
+* An :gdef:`option` generally has a numeric or string value, such as :opt:`Firstorder Depth`.
+* A :gdef:`table` contains a set of strings or qualids.
+* In addition, some commands provide settings, such as :cmd:`Extraction Language`.
+
+.. FIXME Convert "Extraction Language" to an option.
+
+Flags, options and tables are identified by a series of identifiers, each with an initial
+capital letter.
+
+.. cmd:: Set @setting_name {? {| @int | @string } }
+   :name: Set
+
+   .. insertprodn setting_name setting_name
+
+   .. prodn::
+      setting_name ::= {+ @ident }
+
+   If :n:`@setting_name` is a flag, no value may be provided; the flag
+   is set to on.
+   If :n:`@setting_name` is an option, a value of the appropriate type
+   must be provided; the option is set to the specified value.
+
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
+
+   .. warn:: There is no flag or option with this name: "@setting_name".
+
+      This warning message can be raised by :cmd:`Set` and
+      :cmd:`Unset` when :n:`@setting_name` is unknown.  It is a
+      warning rather than an error because this helps library authors
+      produce Coq code that is compatible with several Coq versions.
+      To preserve the same behavior, they may need to set some
+      compatibility flags or options that did not exist in previous
+      Coq versions.
+
+.. cmd:: Unset @setting_name
+   :name: Unset
+
+   If :n:`@setting_name` is a flag, it is set to off.  If :n:`@setting_name` is an option, it is
+   set to its default value.
+
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
+
+.. cmd:: Add @setting_name {+ {| @qualid | @string } }
+
+   Adds the specified values to the table :n:`@setting_name`.
```

Comment:
How does `Add Warnings Append` fit into this?  Is "Append" part of the setting name?

Reply:
I think you mean `Set Warnings Append` (which is a strict equivalent of `Set Warnings` BTW). The answer is that this is the black sheep. Since a recent PR of mine (#11997), it is the last remaining user of `declare_string_option` and the only option that does not support the `Unset` command.